internal/config/config_test.go

```diff
@@ -0,0 +1,175 @@
+// +build envtest
+
+/*
+ Copyright 2021 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/crunchydata/postgres-operator/internal/controller/runtime"
+	"github.com/crunchydata/postgres-operator/internal/naming"
+	"github.com/crunchydata/postgres-operator/pkg/apis/postgres-operator.crunchydata.com/v1beta1"
+	"gotest.tools/v3/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+)
+
+func TestSetContainerImage(t *testing.T) {
+
+	t.Run("env var not set and spec empty", func(t *testing.T) {
+
+		assert.Equal(t, SetContainerImage("", CrunchyPostgresHA_PG10), "")
+	})
+
+	t.Run("image name from spec", func(t *testing.T) {
+
+		os.Setenv(CrunchyPostgresHA_PG10, "envVarImage")
```

Comment:
🔧 Go 1.17 has a `testing.T.Setenv()` that makes sure to undo the effect when the test ends. https://tip.golang.org/doc/go1.17#testing

We should do something similar with a note to replace it with the standard one once we are on Go 1.17.
https://github.com/golang/go/blob/3e48c0381fd1/src/testing/testing.go#L986

```go
func setEnv(t testing.TB, key, value string) {
	t.Helper()
	… os.LookupEnv(key)
	… os.Setenv(key, value)
	t.Cleanup(…)
}

Reply:
:+1: note added. At this point, I'll have to add a call to `os.Unsetenv` on certain tests to make sure I'm accounting for locally set variables. :thinking: I'm not sure if there might be a better option.